atomic_reactor/plugins/pre_yum_proxy.py

```diff
@@ -0,0 +1,162 @@
+        """
+        super(YumProxyPlugin, self).__init__(tasker, workflow)
+
+        # TODO: Is this dir accessbile in the build context?
+        self.workdir = os.path.join(self.workflow.source.workdir, 'atomic-reactor-yum-repos')
+        self.all_yum_repo_files = {}
+        self.updated_yum_repo_files = {}
+        self.yum_proxies = None
+        # TODO: Clean up containers and its volumes.
+        self._container_ids = []
+
+    def run(self):
+        """
+        run the plugin
+        """
+        try:
+            yum_proxies = get_yum_proxies(self.workflow)
+        except KeyError:
+            return
+        # Let's make it easier to search which hosts should be proxied.
+        self.yum_proxies = {info['proxied']: info['proxy'] for info in yum_proxies}
+
+        # Iterate through each parent image to find any hard coded yum repo file.
+        for parent_image_id in self.workflow.builder.parent_images.keys():
+            # TODO: will this ^ ignore SCRATCH_FROM ? Needs testing, but it looks like
+            # SCRATCH_FROM are exluded from parent_images .
+            self.all_yum_repo_files[parent_image_id] = self._get_yum_repos(parent_image_id)
+
+        self._generate_new_yum_repo_files()
+        self._update_dockerfile()
+
+    def _get_yum_repos(self, image_id):
+        # TODO: This needs a retry
+        dest_path = os.path.join(self.workdir, 'original', image_id)
+        try:
+            os.makedirs(dest_path)
+        except OSError:
+            # Ignore if dirs have already been created. Python 2 doesn't suport exists_ok param
+            pass
+        container_id = self.tasker.copy_file(
+            image_id,
+            src_path='/etc/yum.repos.d',
+            dest_path=dest_path,
+            create_kwargs={'user': 'root'},
+        )
+        self._container_ids.append(container_id)
+        return self._parse_yum_repos(dest_path)
+
+    def _parse_yum_repos(self, dest_path):
+        dest_path = os.path.join(dest_path, 'yum.repos.d')
+        yum_repo_files = {}
+        for repo_file_path in os.listdir(dest_path):
+            repo_file_content = ConfigParser()
+            repo_file_content.read(os.path.join(dest_path, repo_file_path))
+            yum_repo_files[repo_file_path] = repo_file_content
+        return yum_repo_files
+
+    def _generate_new_yum_repo_files(self):
+        for parent_image_id, yum_repo_files in self.all_yum_repo_files.items():
+            for yum_repo_file_name, yum_repo_file_content in yum_repo_files.items():
+                rewrite_file = False
+                for yum_repo_name in yum_repo_file_content.sections():
+                    baseurl = yum_repo_file_content.get(yum_repo_name, 'baseurl')
+                    if not baseurl:
+                        continue
+                    parts = urlsplit(baseurl)
+                    if parts.netloc in self.yum_proxies:
+                        # TODO: Do we also need to replace URL schema?
```

Comment:
You mean the scheme, as in 'https', 'http' etc? Hmm, not sure.

Reply:
Yes. Otherwise we'll have to have the internal proxied host to use an identity cert that has been issued by an external entity. If we don't do that, we'll have to bake the internal root CA into the image being built?